### PR TITLE
#164047071 Users should receive notifications

### PR DESCRIPTION
**Pull request by JumaKahiga**
#### Description
___
With users having the ability to follow each other and favorite articles, user experience can be increased by allowing users to get notifications when an author they follow publishes or an article they favorite receives a comment. 

#### Type of Change
___
[x] New Feature

#### How has this been tested
___
[x] Unit tests

#### Checklist
___
[x] I have added tests that prove my feature is working

#### Pivotal Tracker
___
[164047071](https://www.pivotaltracker.com/story/show/164047071)

### CODE REVIEW

File: authors/apps/article/views.py [hunk 1]
```diff
@@ -31,6 +40,13 @@ def post(self, request):
         article = request.data
         article_data = dict(article)
 
+        try:
```
Comment (mnswaleh) [inline authors/apps/article/views.py]: You could reuse
`
        CommentVerification.check_profile(self, request.user.id)
`
to verify if profile exists

Comment (JumaKahiga) [reply authors/apps/article/views.py]: Thank you for that heads-up @mnswaleh

**Review by winniekariuki — COMMENTED** [review]
Nice implementation of the feature but perhaps you can make your docstrings more descriptive in future to aid a reader to understand your code.

File: authors/apps/article/views.py [hunk 2]
```diff
@@ -375,7 +422,7 @@ def check_like(self, serial_data, profile_id, comment_id):
 
         if 'subcomments' in serial_data:
             if len(serial_data['subcomments']) > 0:
-                for return_data in serial_data['subcomments']:
+                for return_data in serial_datsa['subcomments']:
```
Comment (JoanNgatia) [inline authors/apps/article/views.py]: Nitpick fix: typo `serial_datsa`

**Review by JoanNgatia — CHANGES_REQUESTED** [review]

File: authors/apps/article/tests/test_report_article.py [hunk 3]
```diff
@@ -69,6 +69,16 @@ def setUp(self):
         token_request_admin= json.loads(self.request_tkn_admin.content)
         self.token_admin = token_request_admin["user"]["token"]
 
+        create_profile = self.client.post('/api/profile/create_profile/', self.user,
+                                          HTTP_AUTHORIZATION='Token ' + self.token,
+                                          format='json')
+
+        print(create_profile, ">>>>>>>>>")
```
Comment (JoanNgatia) [inline authors/apps/article/tests/test_report_article.py]: You should remove the print statement.

File: authors/apps/article/tests/test_article.py [hunk 4]
```diff
@@ -228,7 +256,7 @@ def test_delete_article(self):
         self.assertEqual(response_article.status_code,
                          status.HTTP_202_ACCEPTED)
 
-    def test_delete_unexistting_article(self):
+    def test_delete_unexisting_article(self):
```
Comment (JoanNgatia) [inline authors/apps/article/tests/test_article.py]: rename this to non-existent

File: authors/apps/profiles/serializers.py [hunk 5]
```diff
@@ -58,15 +61,42 @@ class Meta:
 
 
 class FollowListSerializer(serializers.ModelSerializer):
+	"""Serializes following/followers list."""
```
Comment (JoanNgatia) [inline authors/apps/profiles/serializers.py]: This comment isn't needed as the class name is self-explanatory, in my opinion.

File: authors/apps/profiles/serializers.py [hunk 6]
```diff
@@ -58,15 +61,42 @@ class Meta:
...
 	username = serializers.SlugRelatedField(read_only=True, slug_field='username')
 	follow_since = serializers.ReadOnlyField(source="created_time")
 
 	class Meta:
 		model = UserProfile
 		fields = ["username", "follow_since"]
 
+class NotificationSerializer(serializers.ModelSerializer):
+	"""Serializes notifications list."""
```
Comment (JoanNgatia) [inline authors/apps/profiles/serializers.py]: Edit this comment to differentiate it from the `NotifySerializer`. You can add a comment why the notify serializer had to be different. i.e it's use case

**Review by JoanNgatia — CHANGES_REQUESTED** [review]

**Review by JoanNgatia — APPROVED** [review]

**Pull request merged**